corehq/apps/locations/static/locations/js/widgets.js

```diff
@@ -145,6 +147,50 @@ hqDefine("locations/js/widgets", [
         $select.trigger('select-ready');
     }
 
+    function updateScreenReaderNotification(notificationText, notificationElementId) {
+        const srNotification = $(notificationElementId);
+        if (notificationText) {
+            srNotification.html("<p>" + notificationText + "</p>");
+            srNotification.attr("aria-live","assertive");
+        } else {
+            srNotification.attr("aria-live","off");
+        }
+    }
+
+    function handleLocationWarnings(select) {
+        const requestDomain = initialPageData.get('domain');
+        const selectedLocations = select.val();
+        const requestUserDomainMemberships = initialPageData.get('request_user_domain_memberships');
+        const requestUserLocations = _.find(requestUserDomainMemberships, function (dm) {
+            if (dm.domain === requestDomain) {
+                return dm.assigned_location_ids;
+            }
+        });
+        let shareLocations = false;
```

Comment:
I would have expected `can_access_all_locations` to be part of this calculation. To me it looks like there are two mechanisms that hide the warning `can_access_all_locations` for including the divs in the dom and this to hid the divs if they are present. Is that correct?